### ISSUE TITLE
Xcode preview support for custom views

### ISSUE THREAD

**Issue by merbula** [aws-amplify/amplify-ui-swift-authenticator#63]
Is there a way to get Xcode preview working when using custom views? For example custom login preview gives error: "'SignInState' initializer is inaccessible due to 'internal' protection level"

struct MyCustomLoginView: View {    
    @StateObject var state: SignInState
    
    var body: some View {
        VStack {
        ...
        ...
        }
    }
}

#Preview {
    let state = SignInState(...)
    return MyCustomLoginView(state: state)
}

**Comment by ruisebas** [comment]
Using Xcode previews with custom views is currently not supported, as the states are not meant to be initialized outside the component.

In the meantime, you can workaround this restriction for custom Sign In, Sign Up and Reset Password views by using the `Authenticator` and setting the `initialStep` to one of those. For example:

```swift
#Preview {
    Authenticator(
        initialStep: .signIn,
        signInContent: { state in
            MyCustomLoginView(state: state)
        }
    ) { state in
        Text("Test")
    }
}
```

**Comment by ruisebas** [comment]
This has been addressed in version [1.1.2](https://github.com/aws-amplify/amplify-ui-swift-authenticator/releases/tag/1.1.2) with the introduction of a [`PreviewFactory`](https://github.com/aws-amplify/amplify-ui-swift-authenticator/blob/main/Sources/Authenticator/Utilities/PreviewFactory.swift) utility that provides empty and no-opt states that you can use in previews.

For example:
```swift
import Authenticator

struct CustomSignInView: View {
    @ObservedObject var state: SignInState

    var body: some View { /*...*/ }
}

#Preview {
    CustomSignInView(
        state: PreviewFactory.States.signIn()
    )
}
```
